File: C++/1014. Best Sightseeing Pair.cpp
class Solution {
public:
    int maxScoreSightseeingPair(vector<int>& values) {
        int j= values[0];
        int res= 0;
        for(int i=1;i<values.size();i++){
            res=max(res, j+values[i]-i);
            j=max(j, values[i]+i);
        }
        return res;
    }
};

File: JavaScript/1014. Best Sightseeing Pair.js
/**
 * @param {number[]} values
 * @return {number}
 */
var maxScoreSightseeingPair = function(values) {
    var j= values[0];
    var res= 0;
    for(var i=1;i<values.length;i++){
        res=Math.max(res, j+values[i]-i);
        j=Math.max(j, values[i]+i);
    }
    return res;
};
